Remotion/Validation/UnitTests/Implementation/CompoundValidationMessageFactoryTest.cs

```diff
@@ -14,23 +14,160 @@
+          .Setup(_ => _.CreateValidationMessageForPropertyValidator(validatorStub.Object, propertyStub.Object))
+          .Returns(expectedMessage);
+
+      var factory = new CompoundValidationMessageFactory(new[] { factoryStub.Object });
+
+      var result = factory.CreateValidationMessageForPropertyValidator(validatorStub.Object, propertyStub.Object);
+
+      Assert.That(result, Is.SameAs(expectedMessage));
+    }
+
+    [Test]
+    public void CreateValidationMessageForPropertyValidator_WithMultipleFactories_ContainsMessageFromFirstFactory ()
+    {
+      var validatorStub = new Mock<IPropertyValidator>();
+      var propertyStub = new Mock<IPropertyInformation>();
+
+      var expectedMessage = new InvariantValidationMessage("Oh no, a validation error!");
+
+      var factory1Stub = new Mock<IValidationMessageFactory>();
+      factory1Stub
+          .Setup(_ => _.CreateValidationMessageForPropertyValidator(validatorStub.Object, propertyStub.Object))
+          .Returns(expectedMessage);
+
+      var extraMessage = new InvariantValidationMessage("I sure hope this is not displayed.");
+
+      var factory2Stub = new Mock<IValidationMessageFactory>();
+      factory2Stub
+          .Setup(_ => _.CreateValidationMessageForPropertyValidator(validatorStub.Object, propertyStub.Object))
+          .Returns(extraMessage);
+
+      var factory = new CompoundValidationMessageFactory(new[] { factory1Stub.Object, factory2Stub.Object });
+
+      var result = factory.CreateValidationMessageForPropertyValidator(validatorStub.Object, propertyStub.Object);
+
+      Assert.That(result, Is.EqualTo(expectedMessage));
+    }
+
+    [Test]
+    public void CreateValidationMessageForPropertyValidator_WithFirstFactoryReturningNullMessage_ReturnsMessageFromSecondFactory ()
+    {
+      var validatorStub = new Mock<IPropertyValidator>();
+      var propertyStub = new Mock<IPropertyInformation>();
+
+      var factory1Stub = new Mock<IValidationMessageFactory>();
+      factory1Stub
+          .Setup(_ => _.CreateValidationMessageForPropertyValidator(validatorStub.Object, propertyStub.Object))
+          .Returns((ValidationMessage)null);
+
+      var expectedMessage = new InvariantValidationMessage("Oh no, a validation error!");
+
+      var factory2Stub = new Mock<IValidationMessageFactory>();
+      factory2Stub
+          .Setup(_ => _.CreateValidationMessageForPropertyValidator(validatorStub.Object, propertyStub.Object))
+          .Returns(expectedMessage);
+
+      var factory = new CompoundValidationMessageFactory(new[] { factory1Stub.Object, factory2Stub.Object });
+
+      var result = factory.CreateValidationMessageForPropertyValidator(validatorStub.Object, propertyStub.Object);
+
+      Assert.That(result, Is.EqualTo(expectedMessage));
     }
 
     [Test]
-    [Ignore("RM-5906")]
-    public void CreateValidationMessageForObjectValidator ()
+    public void CreateValidationMessageForObjectValidator_ContainsMessageFromSavedFactory ()
```

Comment:
my mistake: I didn't highlight all occurances of naming updates. Next time, I'll mention it (so long as I don't forget).

=> please check the reminaing tests and apply naming guidelines from the original review